chain/transaction.go

```diff
@@ -174,6 +154,28 @@ type Transaction struct {
 	stateKeys state.Keys
 }
 
+// newTransaction creates a Transaction and initializes the private fields.
+func newTransaction(base *Base, actions Actions, auth Auth) (*Transaction, error) {
```

Comment:
`newTransaction` initializes `bytes` `size` and `id` straight after the codec marshal.